server/src/main/java/com/cloud/api/query/dao/NetworkOfferingJoinDaoImpl.java

```diff
@@ -43,9 +43,12 @@ protected NetworkOfferingJoinDaoImpl() {
     }
 
     @Override
-    public List<NetworkOfferingJoinVO> findByDomainId(long domainId) {
+    public List<NetworkOfferingJoinVO> findByDomainId(long domainId, Boolean includeAllDomainOffering) {
         SearchBuilder<NetworkOfferingJoinVO> sb = createSearchBuilder();
         sb.and("domainId", sb.entity().getDomainId(), SearchCriteria.Op.FIND_IN_SET);
+        if (includeAllDomainOffering) {
+            sb.or("zId", sb.entity().getZoneId(), SearchCriteria.Op.NULL);
+        }
```

Comment:
so if we include "all domain offerings" the zone is allowed to be null?

Reply:
Was copy-paste error, fixed this.
When includeAllDomainOffering is true, it should return offerings which have domainId in DB and also offerings for which domain is not set as they will be available for all domains.